pkg/imagestream/imagestream_test.go

```diff
@@ -0,0 +1,423 @@
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      imageStreamName,
+					Namespace: imageStreamNamespace,
+				},
+				Spec: imagev1.ImageStreamSpec{
+					LookupPolicy: imagev1.ImageLookupPolicy{
+						Local: false,
+					},
+					Tags: []imagev1.TagReference{},
+				},
+			},
+			testImageTag:  imageTag,
+			expectedError: fmt.Errorf("imageStream object network-tools in namespace openshift has no tags"),
+		},
+		{
+			testImageStreamClientWithDummyObject: &imagev1.ImageStream{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      imageStreamName,
+					Namespace: imageStreamNamespace,
+				},
+				Spec: imagev1.ImageStreamSpec{
+					LookupPolicy: imagev1.ImageLookupPolicy{
+						Local: false,
+					},
+					Tags: []imagev1.TagReference{{
+						Name: "latest",
+						From: &corev1.ObjectReference{
+							Kind: "DockerImage",
+							Name: "quay.io/dummy-server/ocp-v4.0-art-dev@sha256:c8bc1d7bdf77538653ff1cd40d9bfa00f0",
+						},
+					}},
+				},
+			},
+			testImageTag: "4.16",
+			expectedError: fmt.Errorf("image tag 4.16 not found for imageStream object network-tools " +
+				"in namespace openshift"),
+		},
+		{
+			testImageStreamClientWithDummyObject: &imagev1.ImageStream{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      imageStreamName,
+					Namespace: imageStreamNamespace,
+				},
+				Spec: imagev1.ImageStreamSpec{
+					LookupPolicy: imagev1.ImageLookupPolicy{
+						Local: false,
+					},
+					Tags: []imagev1.TagReference{{
+						Name: "",
+						From: &corev1.ObjectReference{
+							Kind: "DockerImage",
+							Name: "quay.io/dummy-server/ocp-v4.0-art-dev@sha256:c8bc1d7bdf77538653ff1cd40d9bfa00f0",
+						},
+					}},
+				},
+			},
+			testImageTag:  imageTag,
+			expectedError: fmt.Errorf("imageStream object network-tools in namespace openshift has no DockerImage tag value"),
+		},
+		{
+			testImageStreamClientWithDummyObject: &imagev1.ImageStream{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      imageStreamName,
+					Namespace: imageStreamNamespace,
+				},
+				Spec: imagev1.ImageStreamSpec{
+					LookupPolicy: imagev1.ImageLookupPolicy{
+						Local: false,
+					},
+					Tags: []imagev1.TagReference{{
+						Name: "latest",
+						From: nil,
+					}},
+				},
+			},
+			testImageTag:  imageTag,
+			expectedError: fmt.Errorf("imageStream object network-tools in namespace openshift has no DockerImage value"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+
+		runtimeObjects = append(runtimeObjects, testCase.testImageStreamClientWithDummyObject)
+
+		dummyImageStream := clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects:  runtimeObjects,
+			SchemeAttachers: testSchemes,
+		})
+
+		testImageStreamBuilder := buildValidImageStreamBuilder(dummyImageStream)
+
+		dockerImage, err := testImageStreamBuilder.GetDockerImage(testCase.testImageTag)
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, err)
```

Comment:
since you're already checking that the error and its expected value are equal you can leave out this assertion